File: 151002/Naftolski/Lab5/MessageApp/publisher/src/main/java/by/bsuir/publisher/model/entity/Story.java
package by.bsuir.publisher.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.sql.Date;
import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "tbl_story")
public class Story {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    @JoinColumn(name = "creator_id")
    private Creator creator;

    @Column(name = "title", unique = true, length = 64)
    private String title;

    @Column(name = "content", length = 2048)
    private String content;

    @Column(name = "created")
    private Date created;

    @Column(name = "modified")
    private Date modified;
}

File: 151004/Akimenko/DC_task_05/redis/Publisher/Entity/Db/Story.cs
using Publisher.Entity.Common;
using System.ComponentModel.DataAnnotations;

namespace Publisher.Entity.Db
{
    public class story(string title, string content, DateTime created, DateTime modified) : AbstractEntity
    {
        [MinLength(2)]
        public string Title { get; set; } = title;
        [MinLength(4)]
        public string Content { get; set; } = content;
        public DateTime Created { get; set; } = created;
        public DateTime Modified { get; set; } = modified;
        public ICollection<sticker> stickers { get; set; } = [];
        public int creatorId { get; set; }
        public creator creator { get; set; } = null!;

        public story() : this(string.Empty, string.Empty, DateTime.Now, DateTime.Now) { }
        public story(int id, int creatorId, string title, string content, DateTime created, DateTime modified)
            : this(title, content, created, modified)
        {
            Id = id;
            this.creatorId = creatorId;
            creator = new() { Id = creatorId };
        }
    }
}
